sy-server/migrations/2020-07-23-031356_users/up.sql

```diff
@@ -1,5 +1,49 @@
 CREATE EXTENSION IF NOT EXISTS "uuid-ossp";
 
+CREATE TABLE messages (
+  message_id uuid NOT NULL DEFAULT uuid_generate_v4(),
+  sender_id uuid,
+  recipient_id uuid,
+  message_text VARCHAR,
+  sent_on TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT now(),
+  read_on TIMESTAMP WITH TIME ZONE,
+  PRIMARY KEY(message_id),
+  FOREIGN KEY(sender_id) REFERENCES users(user_id),
+  FOREIGN KEY(recipient_id) REFERENCES users(user_id)
+)
+
+CREATE TABLE tasks (
+  task_id uuid NOT NULL DEFAULT uuid_generate_v4(),
+  owner_id uuid NOT NULL,
+  title CHARACTER VARYING(255),
+  description VARCHAR,
+  weight SMALLINT,
+  created_on TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT now(),
+  due_on TIMESTAMP WITH TIME ZONE,
+  FOREIGN KEY(owner_id) REFERENCES users(user_id),
+  PRIMARY KEY(task_id)
+);
+
+CREATE TABLE completed_tasks (
+  task_id uuid NOT NULL,
+  completed_on TIMESTAMP WITH TIME ZONE,
+  completed_by uuid,
+  FOREIGN KEY(task_id) REFERENCES tasks(task_id),
+  FOREIGN KEY(completed_by) REFERENCES users(user_id)
+)
+
+/*
+ * to get assigned group, check task owner's group *AWKWARD
+ * to get assigned user, search this table
+ * if no assigned user, then it's a group task
+ */
+CREATE TABLE task_assignments (
+  task_id uuid NOT NULL,
+  assigned_to_user uuid,
+  FOREIGN KEY (task_id) REFERENCES tasks(task_id),
+  FOREIGN KEY (assigned_to_user) REFERENCES users(user_id),
+)
+
```

Comment:
Because the task will ever be assigned to one user and one group, there is no need for `task_assignments`  A good use case for this pattern is if you have something like a `shopping_cart` and `shopping_cart_items` where there are many items referencing one cart.  So I would attach this info directly to tasks.  where `assigned_user_id` can be nullable (if null its a group task) and make `group_id` be not null (if we assume that ultimately any task should be at a bare minimum assigned to a group)